pkg/kubelet/network/generic/generic.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generic
+
+import (
+	"errors"
+	"os"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/dockertools"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/network"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/exec"
+	"github.com/golang/glog"
+)
+
+type genericNetworkPlugin struct {
+	host network.Host
+}
+
+const (
+	genericPluginName = "generic"
+	initCmd           = "--init"
+	preLaunchCmd      = "--pre-pod-launch"
+	preDeleteCmd      = "--pre-pod-delete"
+	executable        = "/usr/local/bin/kubelet-plugins/net/generic/generic"
+)
+
+func New() network.NetworkPlugin {
+	return &genericNetworkPlugin{}
+}
+
+func (plugin *genericNetworkPlugin) Init(host network.Host) error {
+	plugin.host = host
+	// call the init script
+	out, err := exec.New().Command(executable, initCmd).CombinedOutput()
```

Comment:
I don't see tests for this, and the usual problem with testst is that exec can't be mocked.  Try pkg/util/exec